File: ShoppingCartApp/ClientApp/src/models/product.ts
export interface Product {
  id: number;
  productCode: string;
  name: string;
  description: string;
  price: number;
  pictureUrl: string;
}

export interface ProductParams {
  orderBy: string;
  searchTerm?: string;
  pageNumber: number;
  pageSize: number;
}

File: ShoppingCartApp/Data/Entities/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ShoppingCart.Data.Entities
{
    [Table("Products")]
    public class Product
    {
        [Key]
        public int Id { get; set; }
        public string ProductCode { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public double Price { get; set; }
        public string PictureUrl { get; set; }
    }
}
